catalog/app/containers/Bucket/CodeSamples/Dir.tsx

```diff
@@ -0,0 +1,85 @@
+import { basename } from 'path'
+
+import dedent from 'dedent'
+import * as React from 'react'
+
+import { docs } from 'constants/urls'
+
+import type { SectionProps } from '../Section'
+
+import Code from './Code'
+
+const TEMPLATES = {
+  PY_INIT: (bucket: string) =>
+    dedent`
+      import quilt3 as q3
+      b = q3.Bucket("s3://${bucket}")
+    `,
+  PY_DOWNLOAD: (path: string, dest: string) => {
+    const pyDest = dest || basename(path)
+    return dedent`
+      # Download [[${docs}/api-reference/bucket#bucket.fetch]]
+      b.fetch("${path}", "./${pyDest}")
+    `
+  },
+  PY_LIST: (path: string) =>
+    dedent`
+      # List files [[${docs}/api-reference/bucket#bucket.ls]]
+      b.ls("${path}")
+    `,
+  CLI_LIST: (bucket: string, path: string) =>
+    dedent`
+      # List files [[https://docs.aws.amazon.com/cli/latest/reference/s3/ls.html]]
+      aws s3 ls "s3://${bucket}/${path}"
+    `,
+  CLI_DOWNLOAD: (bucket: string, path: string, dest: string) => {
+    const cliDest = dest ? `"./${dest}"` : '.'
+    const recursiveFlag = dest ? ` --recursive` : ''
```

Comment:
why is this based on the presence of `dest`? i think it should be based on whether the `path` is a prefix or an object (catalog determines that by the presence of a trailing slash afair)

Reply:
tho this is the `Dir.tsx` file, so i guess it's only meant for directories, and therefore it should always be recursive

Reply:
ok, i'm just confused by the naming. i think if you are to keep this function signature, you should at least add a comment clarifying that `dest` is only present for directories, therefore we can infer `recursiveFlag` from that

Reply:
I started with `CodeSamples/Dir.tsx` and `CodeSamples/File.tsx`, but they have too many things in common, and I decided to group them in one component
But, I guess, it doesn't worth it, and it's better to split them again.

Reply:
> But, I guess, it doesn't worth it, and it's better to split them again.

well, i think just adjusting the naming a little bit for clarity would do, but it's up to you

Reply:
Now the code samples for directory and for file are all in one template each, not interrupted by conditions. And, components doing simple things are simple themselves :)